Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,14 +57,26 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool valid =true;
+
     QStringList sections = line.split(BATDATA_DELIMITER);
 
     QString timeString = sections.at(0);
+
+    if (sections.size() !=3)
+        return false;
+
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    if(!batteryData.time.isValid())
+        return false;
+
+    batteryData.voltage = sections.at(1).toDouble(&valid);
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.current = sections.at(2).toDouble(&valid);
+
+    if (!valid)
+        return false;
```

Comment:
follow style guildlines and show me those nice brackets when doing if/else statements. Also, recommended you don't use one boolean to check to different sections, while it may be more efficient, it isn't as clear

Reply:
Adding on to that, reverse the statement, so `if(value) .... else .....` it isn't a big thing, but the reason for this is that the reader wouldn't have to do mental logic to do read statement. That's usually why you'll see booleans be like `isComplete` or `isValid`, it reduces mental overhead and makes larger ifs much easier to read.